core/templates/services/user-backend-api.service.ts

```diff
@@ -150,6 +164,64 @@ export class UserBackendApiService {
       data: data
     }).toPromise();
   }
+
+  async loadProfileImage(username: string = ''): Promise<ImageFile> {
+    // Artificially load an image from a random exploration where you
+    // uploaded an image. We'll pretend that it is the profile image
+    // that is retrieved from the backend.
+
+    // Test image 1.
+    const image_1 = this.assetsBackendApiService.loadImage(
+      AppConstants.ENTITY_TYPE.EXPLORATION, 'lOZraTUa1x8b',
+      'img_20221108_210419_1pse2cowr9_height_275_width_237.png');
+
+    // Test image 2.
+    const image_2 = this.assetsBackendApiService.loadImage(
+      AppConstants.ENTITY_TYPE.EXPLORATION, 'sIW3TNaiFWQD',
+      'img_20221108_111758_k26g4dj4h7_height_201_width_203.png');
+
+    username = username || await this.getUserInfoAsync().then(
+      userInfo => userInfo.getUsername());
+    // All of this logic is just for testing.
+    if (username === 'eric') {
+      return image_1;
+    } else {
+      return image_2;
+    }
+    // ^^^ Delete everything above this line once merged with Hitesh's changes.
+
+    // Fetch profile image for the current user if not specified.
+    username = username || await this.getUserInfoAsync().then(
+      userInfo => userInfo.getUsername());
+    
+    if (this.profileImageCache !== undefined) {
+      return new ImageFile('profile_image.png', this.profileImageCache);
+    }
+    return this._fetchProfileImage(username);
+  }
+
+  private async _fetchProfileImage(username: string): Promise<ImageFile> {
+    let onResolve!: (_: Blob) => void;
+    let onReject!: () => void;
+    const blobPromise = new Promise<Blob>((resolve, reject) => {
+      onResolve = resolve;
+      onReject = reject;
+    });
+
+    const subscription = this.http.get(
+      this.urlInterpolationService.interpolateUrl(
+        this.assetsBackendApiService.profileImageUrlTemplate,
+        {username: username}), {responseType: 'blob'}
+    ).subscribe(onResolve, onReject);
+
+    try {
+      const blob = await blobPromise;
+      this.profileImageCache = blob;
+      return new ImageFile('profile_image.png', blob);
+    } catch {
+      return Promise.reject('profile_image.png');
+    }  
```

Comment:
NOTE: I'm not sure what the behavior will be like in the backend when the user has not set their profile photo yet. But we may need to add some logic here to see if we need to show the default profile image.